edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/CertificateRenewal.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+
+namespace Microsoft.Azure.Devices.Edge.Hub.Service
+{
+    using System;
+    using System.Threading;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class CertificateRenewal : IDisposable
+    {
+        readonly CancellationTokenSource cts;
+
+        public CancellationToken Token => this.cts.Token;
+
+        public CertificateRenewal(EdgeHubCertificates certificates, ILogger logger)
+        {
+            Preconditions.CheckNotNull(certificates, nameof(certificates));
+            Preconditions.CheckNotNull(logger, nameof(logger));
+
+            TimeSpan timeToExpire = certificates.ServerCertificate.NotAfter - DateTime.UtcNow;
+            if (timeToExpire > TimeSpan.Zero)
```

Comment:
I'd prefer if we leave a small buffer here of a few mins before expiration instead of waiting till it is expired. This is so that clients log any TLS validation errors and so on.